### PR TITLE
Language tweaks for useFieldArray

### PR DESCRIPTION
**Pull request by terebentina**
- Make it clearer that defaultValues is required for every field in the array.
- Provide examples for using field.id as the component key.
- Spelling fixes

I edited this directly on GH so please excuse any formatting issues.
Related to https://github.com/react-hook-form/react-hook-form/issues/3335

### CODE REVIEW

**Review by pmaier983 — APPROVED** [review]
LGTM! thanks for the improvement @terebentina!

File: src/data/en/api.tsx [hunk 1]
```diff
@@ -984,28 +984,33 @@ clearErrors('test.firstName'); // for clear single input error
...
+// ❌ incorrect:
+{fields.map((field, index) => <input key={index} ... />)}
+`} />
             </p>
           </li>
           <li>
             <p>
-              Make sure to set <code>defaultValue</code> to{" "}
-              <code>fields[index]</code>.
+              <code>defaultValue</code> must be set for all inputs to <code>fields[index].<input name></code>. This is required even if you supplied <code>defaultValues</code> in the <code>useForm</code> hook.
```
Comment (pmaier983) [inline src/data/en/api.tsx:1013]: `<input name>` this is this reason the Vercel build is failing. Delete it and also lint the changes you made and you'll be good to go!

**Review by pmaier983 — CHANGES_REQUESTED** [review]
Didnt see the vercel build fail on the first review. Fix my comment and we can merge! Thank you! 😄

File: src/data/en/api.tsx (hunk 1 above)
Comment (pmaier983) [inline src/data/en/api.tsx:1013]:
```suggestion
              <code>defaultValue</code> must be set for all inputs to <code>fields[index].</code>. This is required even if you supplied <code>defaultValues</code> in the <code>useForm</code> hook.
```

**Comment by bluebill1049** [comment]
Thanks @pmaier983 for the review and adjustment. :hear

**Comment by bluebill1049** [comment]
thank you @terebentina for the doc improvement 👍 ❤️

**Comment by bluebill1049** [comment]
oops, early morning mistake... try to hit the merge button but clicked close with comment...

**Pull request merged**